kroxylicious-filters/kroxylicious-oauthbearer-validation/src/test/java/io/kroxylicious/proxy/filter/oauthbearer/OauthBearerValidationFilterTest.java

```diff
@@ -273,11 +268,12 @@ void mustDelayWhenSecondFailedAuthentication() throws Exception {
         filter.onSaslAuthenticateRequest(
                 SaslAuthenticateRequestData.HIGHEST_SUPPORTED_VERSION, new RequestHeaderData(), givenAuthenticateRequest, context);
 
-        // then
-        verify(builder, times(2)).shortCircuitResponse(assertArg(actualResponse -> {
-            assertThat(actualResponse).isInstanceOf(SaslAuthenticateResponseData.class);
-            assertThat(((SaslAuthenticateResponseData) actualResponse).errorCode()).isEqualTo(SASL_AUTHENTICATION_FAILED.code());
-        }));
+        var order = inOrder(builder);
+        order.verify(builder)
+                .shortCircuitResponse(argThat((arg) -> saslAuthenticateResponseMatching(data -> data.errorCode() == SASL_AUTHENTICATION_FAILED.code()).matches(arg)));
```

Comment:
We have these autogenerated matchers that might make for slightly simpler code. WDYT?